
# **Problem Statement**


The dataset is from an ongoing cardiovascular study on residents of the town of Framingham, Massachusetts. The classification goal is to predict whether the patient has a 10-year risk of future coronary heart disease (CHD). The dataset provides the patients’ information. It includes over 4,000 records and 15 attributes.

# **Variables**

Each attribute is a potential risk factor. There are both demographic, behavioral, and medical risk factors.

# **Data Description**

**Demographic**:

• **Sex**: male or female("M" or "F")

• **Age**: Age of the patient;(Continuous - Although the recorded ages have been truncated to whole numbers, the concept of age is continuous)

**Behavioral**

• **is_smoking**: whether or not the patient is a current smoker ("YES" or "NO")

• **Cigs Per Day**: the number of cigarettes that the person smoked on average in one day.(can be considered continuous as one can have any number of cigarettes, even half a cigarette.)

**Medical( history)**
• **BP Meds**: whether or not the patient was on blood pressure medication (Nominal)

• **Prevalent Stroke**: whether or not the patient had previously had a stroke (Nominal)

• **Prevalent Hyp**: whether or not the patient was hypertensive (Nominal)

• **Diabetes**: whether or not the patient had diabetes (Nominal) Medical(current)

• **Tot Chol**: total cholesterol level (Continuous)

• **Sys BP**: systolic blood pressure (Continuous)

• **Dia BP**: diastolic blood pressure (Continuous)

• **BMI**: Body Mass Index (Continuous)

• **Heart Rate**: heart rate (Continuous - In medical research, variables such as heart rate though in fact discrete, yet are considered continuous because of large number of possible values).

• **Glucose**: glucose level (Continuous)

# **Predict variable (desired target)**

• 10-year risk of coronary heart disease CHD(binary: “1”, means “Yes”, “0” means “No”) - DV

# **Now lets import required libraries**

In [2]:
# Importing the libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from scipy import stats
import plotly.express as px
import time
import random 
from sklearn.preprocessing import MinMaxScaler, StandardScaler
from sklearn.model_selection import train_test_split
from pylab import rcParams
from sklearn.linear_model import LinearRegression
from sklearn.tree import DecisionTreeRegressor
from sklearn.ensemble import RandomForestRegressor
from sklearn.svm import SVR
from sklearn.neighbors import KNeighborsRegressor
from sklearn.ensemble import AdaBoostRegressor
from sklearn.ensemble import GradientBoostingRegressor
from xgboost.sklearn import XGBRegressor
import lightgbm as lgb
from sklearn.model_selection import GridSearchCV
from sklearn.model_selection import RandomizedSearchCV
from sklearn import metrics

from shapely.geometry import Point,Polygon
import descartes

from sklearn.metrics import classification_report
from imblearn.over_sampling import SMOTE
from imblearn.under_sampling import RandomUnderSampler
from imblearn.pipeline import Pipeline
from collections import Counter
from sklearn.metrics import confusion_matrix, accuracy_score
from sklearn.metrics import roc_curve,precision_recall_curve
from sklearn.metrics import roc_auc_score,recall_score,precision_score
from sklearn.metrics import cohen_kappa_score

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
# Importing the dataset
dataset = pd.read_csv('/content/drive/My Drive/Cardiovascular-Risk-Prediction Pratik M Gumble/data_cardiovascular_risk.csv')

# **Data Inspection**

In [4]:
# First look
dataset.head()

id  age  education sex is_smoking  cigsPerDay  BPMeds  prevalentStroke  \
0   0   64        2.0   F        YES         3.0     0.0                0   
1   1   36        4.0   M         NO         0.0     0.0                0   
2   2   46        1.0   F        YES        10.0     0.0                0   
3   3   50        1.0   M        YES        20.0     0.0                0   
4   4   64        1.0   F        YES        30.0     0.0                0   

   prevalentHyp  diabetes  totChol  sysBP  diaBP    BMI  heartRate  glucose  \
0             0         0    221.0  148.0   85.0    NaN       90.0     80.0   
1             1         0    212.0  168.0   98.0  29.77       72.0     75.0   
2             0         0    250.0  116.0   71.0  20.35       88.0     94.0   
3             1         0    233.0  158.0   88.0  28.26       68.0     94.0   
4             0         0    241.0  136.5   85.0  26.42       70.0     77.0   

   TenYearCHD  
0           1  
1           0  
2           0  
3           1  
4           0

In [5]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3390 entries, 0 to 3389
Data columns (total 17 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   id               3390 non-null   int64  
 1   age              3390 non-null   int64  
 2   education        3303 non-null   float64
 3   sex              3390 non-null   object 
 4   is_smoking       3390 non-null   object 
 5   cigsPerDay       3368 non-null   float64
 6   BPMeds           3346 non-null   float64
 7   prevalentStroke  3390 non-null   int64  
 8   prevalentHyp     3390 non-null   int64  
 9   diabetes         3390 non-null   int64  
 10  totChol          3352 non-null   float64
 11  sysBP            3390 non-null   float64
 12  diaBP            3390 non-null   float64
 13  BMI              3376 non-null   float64
 14  heartRate        3389 non-null   float64
 15  glucose          3086 non-null   float64
 16  TenYearCHD       3390 non-null   int64  
dtypes: float64(9),

In [6]:
dataset.shape

(3390, 17)

There are 17 columns and 3390 rows in our dataset

# **Data Cleaning**
Before providing data to a model, it is essential to clean the data and treat the nulls, outliers, duplicate data records.

# Convert NaN values

In [7]:
dataset.isnull().sum()

id                   0
age                  0
education           87
sex                  0
is_smoking           0
cigsPerDay          22
BPMeds              44
prevalentStroke      0
prevalentHyp         0
diabetes             0
totChol             38
sysBP                0
diaBP                0
BMI                 14
heartRate            1
glucose            304
TenYearCHD           0
dtype: int64

In [8]:
dataset['cigsPerDay'].fillna(0, inplace=True)
dataset['BPMeds'].fillna(0, inplace=True)
dataset['totChol'].fillna(0, inplace=True)
dataset['BMI'].fillna(0, inplace=True)
dataset['heartRate'].fillna(0, inplace=True)

In [9]:
mean_value=dataset['glucose'].mean()
dataset['glucose'].fillna(value=mean_value, inplace=True)

Number of NaN values in glucose is 304 so we replace with it mean value.

In [10]:
dataset.isnull().sum()

id                  0
age                 0
education          87
sex                 0
is_smoking          0
cigsPerDay          0
BPMeds              0
prevalentStroke     0
prevalentHyp        0
diabetes            0
totChol             0
sysBP               0
diaBP               0
BMI                 0
heartRate           0
glucose             0
TenYearCHD          0
dtype: int64

As number of count of NaN values are not more than 20% so we replace these values with zero instead of mean, median or mode.

In [11]:
dataset = dataset.drop(['id','education'],axis=1)

In [12]:
dataset.shape

(3390, 15)

In [13]:
dataset.describe(include='all')

age   sex is_smoking   cigsPerDay       BPMeds  \
count   3390.000000  3390       3390  3390.000000  3390.000000   
unique          NaN     2          2          NaN          NaN   
top             NaN     F         NO          NaN          NaN   
freq            NaN  1923       1703          NaN          NaN   
mean      49.542183   NaN        NaN     9.010619     0.029499   
std        8.592878   NaN        NaN    11.862839     0.169224   
min       32.000000   NaN        NaN     0.000000     0.000000   
25%       42.000000   NaN        NaN     0.000000     0.000000   
50%       49.000000   NaN        NaN     0.000000     0.000000   
75%       56.000000   NaN        NaN    20.000000     0.000000   
max       70.000000   NaN        NaN    70.000000     1.000000   

        prevalentStroke  prevalentHyp     diabetes      totChol       sysBP  \
count       3390.000000   3390.000000  3390.000000  3390.000000  3390.00000   
unique              NaN           NaN          NaN          NaN         NaN   
top                 NaN           NaN          NaN          NaN         NaN   
freq                NaN           NaN          NaN          NaN         NaN   
mean           0.006490      0.315339     0.025664   234.416814   132.60118   
std            0.080309      0.464719     0.158153    51.453991    22.29203   
min            0.000000      0.000000     0.000000     0.000000    83.50000   
25%            0.000000      0.000000     0.000000   205.000000   117.00000   
50%            0.000000      0.000000     0.000000   234.000000   128.50000   
75%            0.000000      1.000000     0.000000   264.000000   144.00000   
max            1.000000      1.000000     1.000000   696.000000   295.00000   

              diaBP          BMI    heartRate      glucose   TenYearCHD  
count   3390.000000  3390.000000  3390.000000  3390.000000  3390.000000  
unique          NaN          NaN          NaN          NaN          NaN  
top             NaN          NaN          NaN          NaN          NaN  
freq            NaN          NaN          NaN          NaN          NaN  
mean      82.883038    25.688437    75.954867    82.086520     0.150737  
std       12.023581     4.427674    12.041020    23.131808     0.357846  
min       48.000000     0.000000     0.000000    40.000000     0.000000  
25%       74.500000    22.972500    68.000000    72.000000     0.000000  
50%       82.000000    25.365000    75.000000    80.000000     0.000000  
75%       90.000000    27.997500    83.000000    85.000000     0.000000  
max      142.500000    56.800000   143.000000   394.000000     1.000000